src/state_machine/coordinator/mod.rs

```diff
@@ -1053,4 +1076,66 @@ pub mod test {
             assert_ne!(share1.z_i, share2.z_i);
         }
     }
+
+    pub fn bad_signature_share_request<Coordinator: CoordinatorTrait, SignerType: SignerTrait>(
+        num_signers: u32,
+        keys_per_signer: u32,
+    ) {
+        let (mut coordinators, mut signers) =
+            run_dkg::<Coordinator, SignerType>(num_signers, keys_per_signer);
+
+        let msg = "It was many and many a year ago, in a kingdom by the sea"
+            .as_bytes()
+            .to_vec();
+
+        // Start a signing round
+        let signature_type = SignatureType::Frost;
+        let message = coordinators
+            .first_mut()
+            .unwrap()
+            .start_signing_round(&msg, signature_type)
+            .unwrap();
+        assert_eq!(
+            coordinators.first_mut().unwrap().get_state(),
+            State::NonceGather(signature_type)
+        );
+
+        // Send the message to all signers and gather responses by sharing with all other signers and coordinator
+        let (outbound_messages, operation_results) =
+            feedback_messages(&mut coordinators, &mut signers, &[message]);
+        assert!(operation_results.is_empty());
+        assert_eq!(
+            coordinators.first_mut().unwrap().get_state(),
+            State::SigShareGather(signature_type)
+        );
+
+        assert_eq!(outbound_messages.len(), 1);
+        match &outbound_messages[0].msg {
+            Message::SignatureShareRequest(_) => {}
+            _ => {
+                panic!("Expected SignatureShareRequest message");
+            }
+        }
+
+        // add NonceResponses so there's more than the number of signers
+        let mut packet = outbound_messages[0].clone();
+        if let Message::SignatureShareRequest(ref mut request) = packet.msg {
+            for _ in 0..1000 {
+                request
+                    .nonce_responses
+                    .push(request.nonce_responses.first().unwrap().clone());
+            }
+        } else {
+            panic!("failed to match message");
+        }
+
+        // Send the SignatureShareRequest message to all signers and share their responses with the coordinator and signers
+        match feedback_messages_with_errors(&mut coordinators, &mut signers, &[packet]) {
```

Comment:
A nit: you could do something like this here which I think is slightly easier to understand
```rust
    let result = feedback_messages_with_errors(&mut coordinators, &mut signers, &[packet]);
    matches!(result, Err(StateMachineError::Signer(SignerError::InvalidNonceResponse), "Expected invalid nonce response error, got {result:?}");
```

Reply:
Did you mean something like this?  What you posted didn't compile:```
        let result = feedback_messages_with_errors(&mut coordinators, &mut signers, &[packet]);
        if !matches!(result, Err(StateMachineError::Signer(SignerError::InvalidNonceResponse))) {
	    panic!("Should have received signer invalid nonce response error, got {result:?}");
        }
```

Reply:
`cargo fmt` kinda butchered it, but yeah in general `!matches!` is better than a `match` with a single specified variant.